pkg/proto/datadog/trace/idx/span.proto

```diff
@@ -0,0 +1,153 @@
+message ArrayValue {
+	// @gotags: json:"values" msg:"values"
+	repeated AnyValue values = 1;
+}
+
+// a repeated list of KeyValue messages that is needed since `oneof`
+// in AnyValue cannot be `repeated` or `map`
+message KeyValueList {
+	// @gotags: json:"key_values" msg:"key_values"
+	repeated KeyValue keyValues = 1;
+}
+
+message SpanLink {
+   // @gotags: json:"trace_id" msg:"trace_id"
+   bytes traceID = 1;                         // Required.
+   // @gotags: json:"span_id" msg:"span_id"
+   fixed64 spanID = 2;                          // Required.
+   // @gotags: json:"attributes" msg:"attributes"
+   repeated KeyValue attributes = 3;
+   // @gotags: json:"tracestate" msg:"tracestate"
+   uint32 tracestate = 4;                      // Optional. W3C tracestate.
+   // @gotags: json:"flags" msg:"flags"
+   uint32 flags = 5;                           // Optional. W3C trace flags. If set, the high bit (bit 31) must be set.
+}
+
+message SpanEvent {
+  // @gotags: json:"time" msg:"time"
+  fixed64 time = 1; // time is the number of nanoseconds between the Unix epoch and this event.
+  // @gotags: json:"name" msg:"name"
+  uint32 name = 2; // name is this event's name.
+  // attributes is a mapping from attribute key string to any value.
+  // The order of attributes should be preserved in the key/value map.
+  // The supported values match the OpenTelemetry attributes specification:
+  // https://github.com/open-telemetry/opentelemetry-proto/blob/a8f08fc49d60538f97ffabcc7feac92f832976dd/opentelemetry/proto/common/v1/common.proto
+  // @gotags: json:"attributes" msg:"attributes"
+  repeated KeyValue attributes = 3;
+}
+
+
+message Span {
+    // service is the name of the service with which this span is associated.
+    // @gotags: json:"service" msg:"service"
+    uint32 service = 1;
+    // name is the operation name of this span.
+    // @gotags: json:"name" msg:"name"
+    uint32 name = 2;
+    // resource is the resource name of this span, also sometimes called the endpoint (for web spans).
+    // @gotags: json:"resource" msg:"resource"
+    uint32 resource = 3;
+    // spanID is the ID of this span.
+    // @gotags: json:"span_id" msg:"span_id"
+    fixed64 spanID = 4;
+    // parentID is the ID of this span's parent, or zero if this span has no parent.
+    // @gotags: json:"parent_id" msg:"parent_id"
+    uint64 parentID = 5;
+    // start is the number of nanoseconds between the Unix epoch and the beginning of this span.
+    // @gotags: json:"start" msg:"start"
+    fixed64 start = 6;
+    // duration is the time length of this span in nanoseconds.
+    // @gotags: json:"duration" msg:"duration"
+    uint64 duration = 7;
+    // if there is an error associated with this span
+    // @gotags: json:"error" msg:"error"
+    bool error = 8;
+    // meta is a mapping from tag name to tag value for string-valued tags.
+    // @gotags: json:"attributes,omitempty" msg:"attributes,omitempty"
+    repeated KeyValue attributes = 9;
```

Comment:
I recall talking about this at some point, but forgot the exact discussion 😅 

Isn't there a way to preserve the map-like structure of the existing `meta`/`metrics` fields? (eg map of keys to list of values if multiple values per key are needed). I worry about linear data access the repeated field would enforce.

Reply:
Ahh good catch here, I meant to move this to map but missed it!